com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/TimeProfiler.java

```diff
@@ -255,21 +216,20 @@ public static void createReport(Boolean fromEditor) {
                     writer.flush();
                     IOUtils.closeQuietly(writer);
                 }
-                catch (IOException e) {
-                    throw e;
-                }
             }
         } catch (Exception e) {
             throw new RuntimeException(e);
+        } finally {
+            marks.clear();
+            currentScopes.clear();
```

Comment:
Move marks and scope clearing to the place where we write the report, so that once `Bob.invoke` returns, the time profiler is ready for use again.